Add scenario for provider use tutorial (#182)
hashicorp/katakoda: terraform-provider-intro/order-create.md

Now that you have initialized your Terraform workspace, you will create the order using Terraform.

> Add the following to your `main.tf`{{open}} file.

Reviewer: Same problem as in the last step.

> <pre class="file" data-filename="main.tf" data-target="append">

Reviewer 2: I did not know about `data-target="append"`! Works great!

> <pre class="file" data-filename="main.tf" data-target="append">

Reviewer: I think the initial `main.tf` could use an extra newline at the end of it, otherwise you end up with:

```
terraform {
  required_providers {
    hashicups = {
      versions = ["0.2"]
      source = "hashicorp.com/edu/hashicups"
    }
  }
}
provider "hashicups" {
  username = "education"
  password = "test123"
}
# ...
```

Which just looks a little ugly without a blank line between the blocks.

provider "hashicups" {
  username = "education"
  password = "test123"
}

resource "hashicups_order" "edu" {
  items {
    coffee {
      id = 3
    }
    quantity = 2
  }
  items {
    coffee {
      id = 2
    }
    quantity = 2
  }
}

output "edu_order" {
  value = hashicups_order.edu
}
</pre>

> Run `terraform apply` to create the order. Notice how the execution plan shows a proposed order, with additional information about the order.

Reviewer: ```suggestion
Now apply this change to create the order.
```

`terraform apply`{{execute T2}}

> Remember to confirm the apply step with a `yes`{{execute T2}}.

Reviewer: ```suggestion
Notice how the execution plan shows a proposed order, with additional information about the order.

Remember to confirm the apply step with a `yes`{{execute T2}}.
```

Reviewer: Also when testing scenarios, occasionally people accidentally ran the `yes` command (when they clicked on the yes bit before the `terraform apply` bit. I worked around this by adding an alias to coursebase.sh: https://github.com/hashicorp/katakoda/blob/9426a67462da92cb2b7b2120857347bfc4c1c297/terraform-organize-configuration-workshop/courseBase.sh#L9

Once the apply completes, the provider saves the resource's state. If you're running Terraform locally, your `terraform.tfstate` file contains this state. You can also view the state by running `terraform state show <resource_name>`.

Retrieve the order's state.

`terraform state show hashicups_order.edu`{{execute T2}}

### Verify order created

Verify the order was created by retrieving the order details via the API.

> `curl -X GET  -H "Authorization: ${HASHICUPS_TOKEN}" localhost:19090/orders/1`{{execute T2}}

Reviewer 2: piping to `jq` gives me much prettier output and is available by default in the katacoda image
